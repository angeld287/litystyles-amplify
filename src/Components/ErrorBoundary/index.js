import React from "react";
import PropTypes from 'prop-types'
import { BugFilled } from '@ant-design/icons'

class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            error: null,
            errorInfo: null,
        };
    }

    componentDidCatch(error, errorInfo) {
        // Tambi√©n puedes registrar el error en un servicio de reporte de errores
        //logErrorToMyService(error, errorInfo);
        //console.log('componentDidCatch', error, errorInfo);
        this.setState({ hasError: true, error, errorInfo })
    }

    render() {
        if (this.state.hasError) {
            // Puedes renderizar cualquier interfaz de repuesto
            return (
                <>
                    <BugFilled style={{ fontSize: '50px', color: 'red' }} />;
                    <h1>Something went wrong.</h1>
                </>
            );
        }

        return this.props.children;
    }
}

ErrorBoundary.propTypes = {
    children: PropTypes.any
}

export default ErrorBoundary;
