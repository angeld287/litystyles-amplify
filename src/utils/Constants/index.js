import {
    Classes,
    Intent,
} from "@blueprintjs/core";

export const CustomClasses = {
    ...Classes
};

export const CustomIntents = {
    ...Intent
};