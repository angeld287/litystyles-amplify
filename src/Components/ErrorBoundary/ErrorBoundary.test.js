import { render } from '../../utils/test-utils';
import ErrorBoundary from './index';

test('Verify correct renderization of ErrorBoundary Component', () => {
    const { asFragment } = render(<ErrorBoundary><h1>testing Error Boundary</h1></ErrorBoundary>);
    expect(asFragment()).toMatchSnapshot();
})