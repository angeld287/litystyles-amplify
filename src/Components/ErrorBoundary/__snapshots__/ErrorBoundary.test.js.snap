// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify correct renderization of ErrorBoundary Component 1`] = `
<DocumentFragment>
  <h1>
    testing Error Boundary
  </h1>
</DocumentFragment>
`;
